src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,980 @@
+    // if offset is zero and validUntil is null then no DST is used
+    if (dstList[activeDstIndex].offset == 0 && dstList[activeDstIndex].validUntil.IsNull())
+    {
+        return TimeState::kStopped;
+    }
+    if (!dstList[activeDstIndex].validUntil.IsNull() && dstList[activeDstIndex].validUntil.Value() <= chipEpochTime)
+    {
+        mDstOffsetObj.size = 0;
+        VerifyOrReturnValue(mTimeSyncDataProvider.ClearDSTOffset() == CHIP_NO_ERROR, TimeState::kInvalid);
+        // list is empty, generate DSTTableEmpty event
+        sendDSTTableEmptyEvent(GetDelegate()->GetEndpoint());
+        return TimeState::kInvalid;
+    }
+    if (activeDstIndex > 0)
+    {
+        mDstOffsetObj.size    = dstList.size() - activeDstIndex;
+        auto newDstOffsetList = dstList.SubSpan(activeDstIndex);
+        VerifyOrReturnValue(mTimeSyncDataProvider.StoreDSTOffset(newDstOffsetList) == CHIP_NO_ERROR, TimeState::kInvalid);
+        VerifyOrReturnValue(mTimeSyncDataProvider.LoadDSTOffset(mDstOffsetObj) == CHIP_NO_ERROR, TimeState::kInvalid);
+        return TimeState::kChanged;
+    }
+    return TimeState::kActive;
+}
+
+TimeSyncEventFlag TimeSynchronizationServer::GetEventFlag()
+{
+    return mEventFlag;
+}
+
+void TimeSynchronizationServer::ClearEventFlag(TimeSyncEventFlag flag)
+{
+    uint8_t eventFlag = to_underlying(mEventFlag) ^ to_underlying(flag);
+    mEventFlag        = static_cast<TimeSyncEventFlag>(eventFlag);
+}
+
+namespace {
+
+class TimeSynchronizationAttrAccess : public AttributeAccessInterface
+{
+public:
+    // register for the TimeSync cluster on all endpoints
+    TimeSynchronizationAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR ReadTrustedTimeSource(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadDefaultNtp(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadTimeZone(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadDSTOffset(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+    CHIP_ERROR ReadLocalTime(EndpointId endpoint, AttributeValueEncoder & aEncoder);
+};
+
+TimeSynchronizationAttrAccess gAttrAccess;
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadTrustedTimeSource(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    auto tts = TimeSynchronizationServer::Instance().GetTrustedTimeSource();
+    return aEncoder.Encode(tts);
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadDefaultNtp(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    char buffer[DefaultNTP::TypeInfo::MaxLength()];
+    MutableCharSpan dntp(buffer);
+    err = TimeSynchronizationServer::Instance().GetDefaultNtp(dntp);
+    if (err == CHIP_NO_ERROR)
+    {
+        err = aEncoder.Encode(CharSpan(buffer, dntp.size()));
```

Comment:
```suggestion
        err = aEncoder.Encode(dntp);
```
Now that #27441 has merged.  Followup is fine here.

Reply:
Tracked in https://github.com/project-chip/connectedhomeip/issues/27575, resolving this comment for now, further followup on the issue.